src/plugin_discovery/find_plugin_specs.js

```diff
@@ -0,0 +1,134 @@
+import { Observable } from 'rxjs';
+
+import { transformDeprecations, Config } from '../server/config';
+
+import {
+  extendConfigService,
+  disableConfigExtension,
+} from './plugin_config';
+
+import {
+  createPackAtPath$,
+  createPacksInDirectory$,
+} from './plugin_pack';
+
+import {
+  isInvalidDirectoryError,
+  isInvalidPackError,
+} from './errors';
+
+function defaultConfig(settings) {
+  return Config.withDefaultSchema(
+    transformDeprecations(settings)
+  );
+}
+
+function waitForComplete(observable) {
```

Comment:
nit: I couldn't tell what `waitForComplete` was trying to accomplish without finding the function and reading the definition.  Perhaps something like `bufferAllResults` would be clearer?